docs/architecture/decisions/0002-actor-unique-ids.md

```diff
@@ -0,0 +1,25 @@
+# 2. Actor unique IDs
+
+Date: 2024-03-12
+
+## Status
+
+Accepted
+
+## Context
+
+We need a consistent way to refer to each actor on an LPA: donor, certificate provider, attorneys, replacement attorneys and notified people.
+
+This comes from several needs:
+
+- Creating a clear audit trail when an actor makes changes to an LPA (e.g. when the certificate provider signs)
+- Clearly identifying actors across independent services, such as in event-driven messages
+- Ensuring changes to an actor are made against the right person, particularly when there are multiple actors of the same type on one LPA
+
+## Decision
+
+At the point of donor execution, all actors listed on an LPA will be assigned a v4 UUID that will not change past that point. If an actor already has a UUID (because an upstream service has assigned one) then it will not be overwritten.
```

Comment:
should this just say "unique ID" instead of UUID as they are currently being sent from mrlpa to the lpa-store as the URN string, or are we not going to store them like that?

Reply:
Yeah, this is what I'm hoping to clarify a bit with these PRs. My expectation is that the LPA itself (i.e. what MRLPA sends) would have `{"donor": {"uuid": "ABCD"}}`, but `POST /lpas/{uid}/updates` would have `{"author": "urn:opg:poas:mrlpa:users:ABCD"}`.

I guess simplistically the UUID is an _actor_ and the URN is the _user_, but more specifically I don't think there's any benefit adding the URN context to the UUID because I don't think anyone cares who generated it.

Reply:
This confusion points very well to Andrew's point that we should have created ADRs up-front, apologies for confusion!